crates/runtime/src/actor/component/mod.rs

```diff
@@ -377,65 +392,152 @@ where
-                instance.get_export(engine, func_name)
-            else {
-                trace!(
-                    ?instance_name,
-                    func_name,
-                    "instance does not export the parsed function"
-                );
-                continue;
-            };
+
             let instance_name = Arc::clone(&instance_name);
             let func_name = Arc::new(func_name.to_string());
             let target = Arc::clone(&target);
-            if let Err(err) = linker.func_new_async(
-                Arc::clone(&func_name).as_str(),
-                move |mut store, params, results| {
-                    let instance_name = Arc::clone(&instance_name);
-                    let func_name = Arc::clone(&func_name);
-                    let target = Arc::clone(&target);
-                    let func = func.clone();
-                    Box::new(async move {
-                        let params: Vec<_> = zip(params, func.params())
-                            .map(|(val, ty)| to_wrpc_value(&mut store, val, &ty))
-                            .collect::<anyhow::Result<_>>()
-                            .context("failed to convert wasmtime values to wRPC values")?;
-                        let handler = &store.data().handler;
-                        let target = handler
-                            .identify_interface_target(&target)
-                            .await
-                            .context("failed to identify interface target")?;
-                        let result_values = handler
-                            .call(target, &instance_name, &func_name, params)
-                            .await
-                            .context("failed to call target interface")?;
-                        for (i, (val, ty)) in zip(result_values, func.results()).enumerate() {
-                            let val = from_wrpc_value(&mut store, val, &ty)?;
-                            let result = results.get_mut(i).context("invalid result vector")?;
-                            *result = val;
+            let rpc_name = wrpc_introspect::rpc_func_name(ty);
+            let (param_types, result_types) = match comp_ty.get_import(engine, &ty.name) {
+                Some(types::ComponentItem::ComponentFunc(i)) => (
+                    i.params().collect::<Vec<_>>().into_iter(),
+                    i.results().collect::<Vec<_>>().into_iter(),
+                ),
+                Some(_) | None => {
+                    error!(
+                        "Function param types not found for instance: {} and name: {}",
+                        instance_name, func_name
+                    );
+                    continue;
+                }
+            };
+
+            let result_types_vec: Vec<_> = result_types.collect();
+            let result_types_arc = Arc::new(result_types_vec);
+
+            if let Err(err) = linker.func_new_async(rpc_name, move |mut store, params, results| {
```

Comment:
I think we should be using https://bytecodealliance.github.io/wrpc/wrpc_runtime_wasmtime/fn.polyfill.html here
or, at least, `polyfill_function`, which I've extracted in https://github.com/bytecodealliance/wasmtime/pull/8737.
I'll work on pulling in changes from that PR back to `wrpc-runtime-wasmtime`

Reply:
See https://github.com/bytecodealliance/wrpc/pull/149